assignments/this.js

```diff
@@ -1,26 +1,53 @@
 /* The for principles of "this";
 * in your own words. explain the four principle for the "this" keyword below.
 *
-* 1. 
-* 2. 
-* 3. 
-* 4. 
+* 1. When you use this in a global aspect, it will refer to the window or console.
+* 2. When using methods on an object, the object before the .method is what this will refer to.
+* 3. When an object is created through a constructor function, the this keyword refers to the object returned by the constructor.
+* 4. See example below.
 *
 * write out a code example of each explanation above
 */
 
 // Principle 1
 
 // code example for Window Binding
-
+console.log(this);
 // Principle 2
 
 // code example for Implicit Binding
-
+const implicit = {
+    example: "This is an example",
+    method: function(){
+        console.log(this.example);
+    }
+};
+implicit.method();
 // Principle 3
 
 // code example for New Binding
-
+function Person(name,age){
+    (this.name = name), (this.age = age);
```

Comment:
You have some interesting syntax here. It seems to be working, but the standard syntax, at least to my understanding, would be something like this:

```js
function Person(name, age) {
    this.name = name;
    this.age = age;
}
```

❓ Can you explain to me a little more about the syntax here, like the paraenthesis around the assignments here?

Reply:
Yeah, the examples I've seen included this syntax, and it was just me more or less experimenting and seeing what worked. I guess one thing is that it lets me use one line.